gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/HivePartitionVersionPolicy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance;
+
+import gobblin.configuration.State;
+
+
+/**
+ * Abstract class for selection policies corresponding to {@link HivePartitionVersion}
+ *
+ * @author adsharma
+ */
+public abstract class HivePartitionVersionPolicy implements Policy<HivePartitionVersion> {
+  protected State state;
+  protected HivePartitionDataset dataset;
+
+  public HivePartitionVersionPolicy(State state, HivePartitionDataset dataset) {
```

Comment:
It is weird that this abstract class only has this constructor. Is `State` necessary? Can it be `Properties` or `Config`? I think what this class really needs is a `getHivePartitionDataset()` method.  What do you think?